### DISCUSSION TITLE
How to have multiple services behind one url

### DISCUSSION THREAD

**Discussion by Mortana89** [microsoft/aspire#1653] [Q&A]
Hi,

We have +- 20 microservices, running in Service Fabric. Playing around with Aspire, I was wondering if we could in some way omit our Ocelot gateway (since in the cloud we will replace it by APIM) and configure aspire in such a way that calls via frontend to localhost/API/svc1/controller go to the svc1 endpoints, and calls to localhost/API/svc2/... Go to the svc2 endpoints? This is what ocelot currently does for us, linked to Service Fabric service discovery.

This way we can still use our current working approach (we go to the cloud hosted staginf frontend, set a session var to connect to localhost instead of cloud Dev APIs, and can start debugging), without changing urls everywhere.

Thanks!

**Comment by DamianEdwards** [comment]
Aspire doesn't include an automatic L7 reverse proxy today, which is what would be required to enable what you're talking about (forwarding to different projects based on HTTP request path). There's some more discussion about this including other approaches in https://github.com/dotnet/aspire/issues/836.
